### PR TITLE
CDSR-801: refactor claim answer model

### PR DESCRIPTION
**Pull request by gabor-dorcsinecz**

### CODE REVIEW

File: test/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerServiceSpec.scala [hunk 1]
```diff
@@ -977,7 +967,7 @@ class ClaimTransformerServiceSpec extends AnyWordSpec with Matchers with MockFac
         val acc14                                                            = getAcc14Response()
         val displayDeclaration                                               = sample[DisplayDeclaration].copy(displayResponseDetail = acc14)
         val claim                                                            = getClaimAmounts()
-        val completeClaimsAnswer                                             = sample[CompleteClaimsAnswer].copy(List(claim))
+        val completeClaimsAnswer                                             = ClaimsAnswer(claim)
```
Comment (mikail-khan) [inline test/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerServiceSpec.scala:970]: might be worth renaming this as we don't have this incomplete/complete concept anymore

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/CompleteClaim.scala [hunk 2]
```diff
@@ -62,7 +63,7 @@ object CompleteClaim {
     maybeDuplicateDisplayDeclaration: Option[DisplayDeclaration],
     importerEoriNumber: Option[CompleteImporterEoriNumberAnswer],
     declarantEoriNumber: Option[CompleteDeclarantEoriNumberAnswer],
-    completeClaimsAnswer: CompleteClaimsAnswer
+    completeClaimsAnswer: ClaimsAnswer
```
Comment (mikail-khan) [inline app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/CompleteClaim.scala:66]: rename completeClaimsAnswer to claimsAnswer - should it be claimAnswers or claimsAnswer?

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/Claim.scala [hunk 3]
```diff
@@ -31,8 +32,8 @@ final case class Claim(
 )
 
 object Claim {
-  implicit class ListClaimOps(private val claims: List[Claim]) {
-    def total: BigDecimal = claims.map(c => c.claimAmount).sum
+  implicit class ListClaimOps(private val claims: NonEmptyList[Claim]) {
+    def total: BigDecimal = claims.map(c => c.claimAmount).toList.sum
```
Comment (mikail-khan) [inline app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/Claim.scala:36]: do we use this in the BE? it was added in the FE so calculate the totals in the summary page

Comment (gabor-dorcsinecz) [reply app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/Claim.scala:36]: Yes we do: ClaimTransformerService line 82 and 156

File: test/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerServiceSpec.scala (hunk 1 above)
Comment (gabor-dorcsinecz) [reply test/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerServiceSpec.scala]: done

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/CompleteClaim.scala (hunk 2 above)
Comment (gabor-dorcsinecz) [reply app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/CompleteClaim.scala]: done

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/package.scala [hunk 4]
```diff
@@ -19,13 +19,16 @@ package uk.gov.hmrc.cdsreimbursementclaim
...
 import play.api.libs.json.{Format, JsonValidationError, Reads, Writes}
+import uk.gov.hmrc.cdsreimbursementclaim.models.claim.Claim
 
 package object models {
 
   type Validation[A] = ValidatedNel[String, A]
 
   def invalid[A](error: String): Validation[A] = Invalid(NonEmptyList.one(error))
 
+  type ClaimsAnswer = NonEmptyList[Claim]
```
Comment (aneksamun) [inline app/uk/gov/hmrc/cdsreimbursementclaim/models/package.scala:30]: @gabor-dorcsinecz please can you move it to `answers` package

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/CompleteClaimAnswer.scala [hunk 5]
```diff
@@ -16,12 +16,11 @@
 
 package uk.gov.hmrc.cdsreimbursementclaim.models.claim
 
-import play.api.libs.json.{Json, OFormat}
+import cats.data.NonEmptyList
 
-final case class CompleteClaimAnswer(
-  claims: List[Claim]
-)
+object ClaimsAnswer {
```
Comment (aneksamun) [inline app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/CompleteClaimAnswer.scala:21]: @gabor-dorcsinecz please can you move it to `answers` package

Comment (gabor-dorcsinecz) [reply app/uk/gov/hmrc/cdsreimbursementclaim/models/claim/CompleteClaimAnswer.scala]: Done

File: app/uk/gov/hmrc/cdsreimbursementclaim/models/package.scala (hunk 4 above)
Comment (gabor-dorcsinecz) [reply app/uk/gov/hmrc/cdsreimbursementclaim/models/package.scala]: Done

**Review by aneksamun — APPROVED** [review]

**Pull request merged**